File: bot/commands/reportTaskKillCommand.py

from discord.raw_models import RawReactionActionEvent
from apiClient import ApiClient
from command_spec import PhraseRepository, ReactionCommand


class ReportTaskKillCommand(ReactionCommand):
    def __init__(self, phraseRepository: PhraseRepository, apiClient: ApiClient) -> None:
        super().__init__(phraseRepository)
        self.apiClient = apiClient

    def isMatchTo(self, reaction: str, messageId: str) -> bool:
        response = self.apiClient.get_report_message(messageId)

        if(response.status_code == 200):
            return (
                reaction == self.phraseRepository.get('task_kill_stamp')
            )
        else:
            return False

    def executeForAdd(self, reaction: RawReactionActionEvent) -> None:
        print("report task kill")
        self.apiClient.report_task_kill(reaction.message_id, reaction.user_id)

    def executeForRemove(self, reaction: RawReactionActionEvent) -> None:
        print("cancel task kill")
        self.apiClient.cancel_task_kill(reaction.message_id, reaction.user_id)

File: bot-js/src/commands/ReportTaskKillCommand.ts
import { MessageReaction, User } from 'discord.js';
import { ApiClient } from '../backend/ApiClient';
import { PhraseRepository } from '../support/PhraseRepository';
import { ReactionCommand } from './ReactionCommand';

export class ReportTaskKillCommand implements ReactionCommand {
    constructor(
        private phraseRepository: PhraseRepository,
        private apiClient: ApiClient,
    ) {
    }

    async isMatchTo(reaction: MessageReaction): Promise<boolean> {
        return await this.apiClient.hasReportMessage(reaction.message.id)
            && reaction.emoji.toString() === this.phraseRepository.get("task_kill_stamp");
    }

    async executeForAdd(reaction: MessageReaction, user: User): Promise<void> {
        console.log("report task kill");

        await this.apiClient.reportTaskKill(reaction.message.id, user.id);
    }

    async executeForRemove(reaction: MessageReaction, user: User): Promise<void> {
        console.log("cancel task kill");

        await this.apiClient.cancelTaskKill(reaction.message.id, user.id);
    }
}